hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java

```diff
@@ -0,0 +1,296 @@
+import com.microsoft.azure.management.keyvault.implementation.VaultInner;
+import com.microsoft.azure.management.msi.implementation.IdentityInner;
+import com.microsoft.azure.management.msi.implementation.MSIManager;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.implementation.ResourceManager;
+import com.microsoft.azure.management.storage.Kind;
+import com.microsoft.azure.management.storage.StorageAccount;
+import com.microsoft.azure.management.storage.StorageAccountSkuType;
+import com.microsoft.azure.management.storage.implementation.StorageManager;
+import com.microsoft.rest.RestClient;
+import org.assertj.core.util.Lists;
+
+import java.util.List;
+import java.util.UUID;
+
+import static com.microsoft.azure.arm.core.TestBase.generateRandomResourceName;
+import static com.microsoft.azure.arm.core.TestBase.isRecordMode;
+import static com.microsoft.azure.management.hdinsight.v2018_06_01_preview.scenariotests.HDInsightManagementTestBase.*;
+import static com.microsoft.azure.management.storage.Kind.STORAGE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Manage Azure resources used during test execution
+ */
+public class HDInsightTestResourceManager {
+
+    /**
+     * Default key vault permissions. Give the service principal all permissions
+     */
+    private static final Permissions DEFAULT_PERMISSIONS = new Permissions()
+        .withKeys(Lists.newArrayList(KeyPermissions.values()))
+        .withSecrets(Lists.newArrayList(SecretPermissions.values()))
+        .withCertificates(Lists.newArrayList(CertificatePermissions.values()))
+        .withStorage(Lists.newArrayList(StoragePermissions.values()));
+
+    private static final int AZURE_RESOURCE_NAME_MAX_LEN = 24;
+
+    /**
+     * Entry point to Azure Resources resource management.
+     */
+    private final ResourceManager resourceManager;
+
+    /**
+     * Entry point to Azure Storage resource management.
+     */
+    private final StorageManager storageManager;
+
+    /**
+     * Entry point to Azure Managed Service Identity (MSI) resource management.
+     */
+    private final MSIManager msiManager;
+
+    /**
+     * Entry point to Azure KeyVault resource management.
+     */
+    private final KeyVaultManager keyVaultManager;
+
+    /**
+     * Key vault client.
+     */
+    private final KeyVaultClient keyVaultClient;
+
+    private final String tenantId;
+
+    public HDInsightTestResourceManager(RestClient restClient, String subscription, String tenant) {
+        resourceManager = ResourceManager.authenticate(restClient).withSubscription(subscription);
+        storageManager = StorageManager.authenticate(restClient, subscription);
+        msiManager = MSIManager.authenticate(restClient, subscription);
+        keyVaultManager = KeyVaultManager.authenticate(restClient, tenant, subscription);
+        keyVaultClient = new KeyVaultClient(createKeyVaultRestClient(restClient));

```

Comment:
Does the client access the real KeyVault service?

Reply:
Yes, in record mode, the client access the real KeyVault service to get HTTP response.